test/zenfs_metazone_rollover_test.cc

```diff
@@ -31,35 +34,69 @@ int test_metadata_rollover() {
   return 0;
 }
 
-int test_snapshot_zone_allocation() {
-  std::shared_ptr<Logger> logger;
+int test_mkfs() {
   Status s;
+  DIR* aux_dir;
+  std::shared_ptr<Logger> logger;
+
+  if (FLAGS_aux_path.empty()) {
+      fprintf(stderr, "You need to specify --aux_path\n");
+      return 1;
+  }
+
+  aux_dir = opendir(FLAGS_aux_path.c_str());
+  if (ENOENT != errno) {
+      fprintf(stderr, "Error: aux path exists\n");
+      closedir(aux_dir);
+      return 1;
+  }
 
   s = Env::Default()->NewLogger(GetLogFilename(FLAGS_zbd), &logger);
   if (!s.ok()) {
-    fprintf(stderr, "ZenFS: Could not create logger");
+      fprintf(stderr, "ZenFS: Could not create logger");
   } else {
-    logger->SetInfoLogLevel(DEBUG_LEVEL);
+      logger->SetInfoLogLevel(DEBUG_LEVEL);
   }
 
   ZonedBlockDevice *zbd = zbd_open(false, logger);
-  if (zbd == nullptr) return 1;
+  if (zbd == nullptr) {
+      fprintf(stderr, "Failed to open file\n");;
+      return 1;
+  }
 
-  auto meta_snapshot_zones = zbd->GetMetaSnapshotZones();
-  
-#ifdef WITH_ZENFS_ASYNC_METAZONE_ROLLOVER
+  /* verify if snapshot zones being allocated properly */
+  auto snapshot_zones = zbd->GetSnapshotZones();
   const int zenfs_snapshot_zones_num = 2;
-  if (meta_snapshot_zones.size() != zenfs_snapshot_zones_num) {
-    fprintf(stderr, "Error! Failed to allocate meta snapshot zone");
+  if (snapshot_zones.size() != zenfs_snapshot_zones_num) {
+    fprintf(stderr, "Error! Failed to allocate snapshot zone");
     return 1;
   }
-#else
-  if (meta_snapshot_zones.size() == 0) {
-    std::cout << "Success! Did not allocate meta snapshot zone\n";
-    return 0;
+
+  ZenFS *zenFS;
+  zenFS = new ZenFS(zbd, FileSystem::Default(), logger);
+
+  if (FLAGS_aux_path.back() != '/') FLAGS_aux_path.append("/");
+
+  s = zenFS->MkFS(FLAGS_aux_path, FLAGS_finish_threshold,
+                  FLAGS_max_open_zones, FLAGS_max_active_zones);
+  if (!s.ok()) {
+    fprintf(stderr, "Failed to create file system, error: %s\n",
+            s.ToString().c_str());
+    delete zenFS;
+    return 1;
   }
-#endif
 
+  /* verify if the fist snapshot zone has an empty snapshot record */
+  assert(!snapshot_zones[0].IsEmpty());
```

Comment:
also question here: how do we test if we have a valid and empty snapshot in this snapshot zone.